MetaMask/metamask-extension
# rely upon gas fee controller for gas price estimates

Changes
* returns `false` from getNetworkEIP1559Compatibility if the `SHOW_EIP_1559_UI` environment variable isn't true. This allows testnets such as Ropsten, Goerli, Rinkeby, etc that have launched EIP-1559 to still get good `gasPrice` estimates until we release our UI elements.
* Plugs in values from the `GasFeeController` state into the custom-gas selectors. These selectors will be removed in favor of directly accessing the underlying state values (or creating new, more apt named selectors) in a future PR.
* updates tests to have mock data appropriate for these changes. I attempted to only modify test mock data or remove tests that were no longer relevant (covering removed code)
* Adds calls to start polling on the gasFeeController from a number of places that were previously calling the fetch method from gas.duck.js.
* Removes all fetch related code from gas.duck.js, it now ONLY exists to store custom data which will also go away as we get closer to merging our EIP-1559 work. 

Items remaining
- [x] The gasPrice on send page is being constantly overwritten with values. I need to tap into the customGasPrice state to see if it has been set and then avoid changing gasPrice once that happens. We'll need to figure this out for EIP-1559 too because the maxFeePerGas/gasPrice will update periodically but **isn't visible to the user**, resulting in possibly randomly displaying insufficient funds warnings? 
- [x] Failing test that I'm not sure why is happening. 


### QA / Testing Steps
- Use the send flow on various networks and ensure a reasonable gasPrice is set. this is 1 on rinkeby, 5 on Goerli, 10 on Kovan, 2 on Ropsten (as of this writing 7/14/2021 1:20 CST).
- On the send flow, on custom networks, switch networks and see that the gasPrice updates within a few seconds (no more than 10)
- On the send flow, if you change your gasPrice, your price does not change every 10 seconds.
- Confirmation edit gas price logic works on mainnet (I *believe* that custom networks do not use estimates they set the gasPrice to the transaction gasPrice, the only place estimates are displayed is in the buttons)
- speed up/cancel works

### Known Issues
<details>
<summary>Gas customization modal on mainnet does not update gasPrice values when estimates change</summary>

https://user-images.githubusercontent.com/4448075/125673410-f903853e-933d-4f5b-afb9-5cb81fe72ae7.mov

<br/>
<ul>
<li>I don't know how much time we should spend resolving this as it doesn't actually change the *result* of the behavior from mainnet. If the user had opened this window the gasPrices wouldn't have updated but it doesn't mean the expectations of the network hadn't changed.</li>
<li>Fixing this will require introducing a whole layer of complexity to the existing gas modal that we intend on deprecating soon</li>
</ul>


</details>
